src/app/query/bitcoin/ordinals/use-inscription-by-param.query.ts

```diff
@@ -0,0 +1,35 @@
+import { useQuery } from '@tanstack/react-query';
+
+import { Paginated } from '@shared/models/api-types';
+import { Inscription } from '@shared/models/inscription.model';
+
+import { AppUseQueryConfig } from '@app/query/query-config';
+
+const inscriptionQueryOptions = {
+  staleTime: Infinity,
+  cacheTime: Infinity,
+} as const;
+
+function fetchInscriptionByParam() {
+  return async (param: string) => {
+    const res = await fetch(`https://api.hiro.so/ordinals/v1/inscriptions?${param}`);
+    if (!res.ok) throw new Error('Error retrieving inscription metadata');
+    const data = await res.json();
+    return data as Paginated<Inscription[]>;
+  };
+}
+
+type FetchInscriptionResp = Awaited<ReturnType<ReturnType<typeof fetchInscriptionByParam>>>;
+
+export function useGetInscriptionByParamQuery<T extends unknown = FetchInscriptionResp>(
+  param: string,
+  options?: AppUseQueryConfig<FetchInscriptionResp, T>
+) {
```

Comment:
Don't think it's the best API choice to pass in params themselves. This kinda leaks the implementation details of the API elsewhere in the app. 

Can't we pass in the txid and offset, and form the param internally to this hook?

Reply:
hm, I created separate hook to handle case when we need to get info by output `use-inscription-by-output`, which uses this query, assuming that we prob will need to get information in other part of the app with other param

Reply:
if we will handle it internally in the query => to fetch with other param we will need create same query again?